apps/metadata_playground/app.py

```diff
@@ -0,0 +1,177 @@
+
+import etl.grapher_model as gm
+from etl import paths
+from etl.db import get_engine
+
+###################################################
+# Initial configuration ###########################
+###################################################
+# Set page config
+# st.title("Metadata v2 preview")
+st.set_page_config(page_title="Metadata v2 preview", layout="wide", page_icon="🎨")
+st.title("Metadata v2 preview")
+
+# Current directory
+CURRENT_DIR = Path(__file__).parent.absolute()
+
+# Load metadata from Snapshot
+PATH_METADATA_SNAPSHOT = paths.SNAPSHOTS_DIR / "dummy" / "2020-01-01" / "dummy.csv.dvc"
+with open(PATH_METADATA_SNAPSHOT, 'r') as f:
+    METADATA_SNAPSHOT_BASE = f.read()
+SNAPSHOT_META_TOKEN_SPLIT = "outs:"
+METADATA_SNAPSHOT_DISPLAY = METADATA_SNAPSHOT_BASE.split(SNAPSHOT_META_TOKEN_SPLIT)[0]
+METADATA_SNAPSHOT_EXTRA = METADATA_SNAPSHOT_BASE.split(SNAPSHOT_META_TOKEN_SPLIT)[1]
+# Load metadata from Garden
+PATH_METADATA_GARDEN = paths.STEP_DIR / "data" / "garden" / "dummy" / "2020-01-01" / "dummy.meta.yml"
+with open(PATH_METADATA_GARDEN, 'r') as f:
+    METADATA_GARDEN_BASE = f.read()
+METADATA_GARDEN_DISPLAY = METADATA_GARDEN_BASE
+
+# Catalog path
+CATALOG_PATH = "grapher/dummy/2020-01-01/dummy/dummy"
+
+
+# Functions
+def run_steps() -> None:
+    """Run dummy steps from ETL.
+
+    The environment is assumed to be set.
+    """
+    # env_path = paths.BASE_DIR / ".env.X"
+    # subprocess.run(f"export $(cat {env_path} | xargs)", shell=True)
+    subprocess.run(["poetry", "run", "etl", "dummy", "--grapher"])
+
+
+def get_variable_id() -> int:
+    """Get variable ID"""
+    with Session(get_engine()) as session:
+        variable = session.exec(select(gm.Variable).where(gm.Variable.catalogPath == CATALOG_PATH)).one()
+    return variable.id
+
+
+def get_data_page_url() -> str:
+    """Get data page URL"""
+    HOST = os.environ.get("DB_HOST")
+    VARIABLE_ID = get_variable_id()
+    url = f"http://{HOST}/admin/datapage-preview/{VARIABLE_ID}"
+    return url
+
+
+def reset_metadata_files():
+    """Reset metadata files"""
+    # Reset YAML files
+    with open(PATH_METADATA_SNAPSHOT, 'w') as f:
+        f.write(METADATA_SNAPSHOT_BASE)
+    with open(PATH_METADATA_GARDEN, 'w') as f:
+        f.write(METADATA_GARDEN_BASE)
+
+
+###################################################
+# Show header #####################################
+###################################################
+URL_METADATA = "https://www.notion.so/owid/Metadata-guidelines-29ca6e19b6f1409ea6826a88dbb18bcc"
+
+# Define columns
+col1, col2 = st.columns(2)
+
+# Some info, wrap option
+with col1:
+    # Link to metadata v2 docs
+    st.markdown(f"📚 [Metadata v2 documentation]({URL_METADATA})")
+    WRAP = st.checkbox("Wrap enabled", value=False)
+
+    # Explanation
+    with st.expander("How does this work?"):
+        st.markdown("""
+            This tool lets you visualise how your metadata edits are reflected in a data page.
```

Comment:
I'd emphasise that you need a running admin (and the preferred way is to run this against staging server). When you don't have an admin, you get redirected to a broken page with no explanation.